storage/sectors.go

```diff
@@ -229,52 +204,54 @@ func (m *Miner) onSectorUpdated(ctx context.Context, update sectorUpdate) {
 
 	switch update.newState {
 	// Happy path
-	case api.Packing:
+	case Packing:
 		m.handleSectorUpdate(ctx, sector, m.handlePacking)
-	case api.Unsealed:
+	case Unsealed:
 		m.handleSectorUpdate(ctx, sector, m.handleUnsealed)
-	case api.PreCommitting:
+	case PreCommitting:
 		m.handleSectorUpdate(ctx, sector, m.handlePreCommitting)
-	case api.PreCommitted:
+	case PreCommitted:
 		m.handleSectorUpdate(ctx, sector, m.handlePreCommitted)
-	case api.Committing:
+	case Committing:
 		m.handleSectorUpdate(ctx, sector, m.handleCommitting)
-	case api.CommitWait:
+	case CommitWait:
 		m.handleSectorUpdate(ctx, sector, m.handleCommitWait)
-	case api.Proving:
+	case Proving:
 		// TODO: track sector health / expiration
 		log.Infof("Proving sector %d", update.id)
 
 	// Handled failure modes
-	case api.SealFailed:
+	case SealFailed:
 		log.Warnf("sector %d entered unimplemented state 'SealFailed'", update.id)
-	case api.PreCommitFailed:
+	case PreCommitFailed:
 		log.Warnf("sector %d entered unimplemented state 'PreCommitFailed'", update.id)
-	case api.SealCommitFailed:
+	case SealCommitFailed:
 		log.Warnf("sector %d entered unimplemented state 'SealCommitFailed'", update.id)
-	case api.CommitFailed:
+	case CommitFailed:
 		log.Warnf("sector %d entered unimplemented state 'CommitFailed'", update.id)
 
-		// Faults
-	case api.Faulty:
-		m.handleSectorUpdate(ctx, sector, m.handleFaulty)
-	case api.FaultReported:
-		m.handleSectorUpdate(ctx, sector, m.handleFaultReported)
-
 	// Fatal errors
-	case api.UndefinedSectorState:
+	case UndefinedSectorState:
 		log.Error("sector update with undefined state!")
-	case api.FailedUnrecoverable:
+	case FailedUnrecoverable:
 		log.Errorf("sector %d failed unrecoverably", update.id)
 	default:
 		log.Errorf("unexpected sector update state: %d", update.newState)
 	}
 }
 
 func (m *Miner) AllocatePiece(size uint64) (sectorID uint64, offset uint64, err error) {
-	if padreader.PaddedSize(size) != size {
-		return 0, 0, xerrors.Errorf("cannot allocate unpadded piece")
-	}
+	// TODO: padreader.PaddedSize(1016) == 1016, which seems like a bug to me
```

Comment:
This makes sense to me. `PaddedSize(1000)`, `PaddedSize(1016)`, `PaddedSize(1024)` are all going to round up to 1024 and then take the user bytes to get 1016. So this is a reasonable test to ensure `size` is a valid pieces size (i.e. the user bytes from some power of 2).

Reply:
I agree the naming is a little confusing, it should really be `PaddedUserByteSize` or `PaddedPieceSize`. But it's always going to pad to a power of two and then subtract.

Reply:
sounds good, thank you